SamsungPay/src/main/java/com/braintreepayments/api/SamsungPayClient.java

```diff
@@ -0,0 +1,235 @@
+import static com.samsung.android.sdk.samsungpay.v2.SpaySdk.SPAY_NOT_SUPPORTED;
+import static com.samsung.android.sdk.samsungpay.v2.SpaySdk.SPAY_READY;
+
+public class SamsungPayClient {
+
+    private final BraintreeClient braintreeClient;
+
+    @VisibleForTesting
+    SamsungPayInternalClient internalClient;
+
+    public SamsungPayClient(BraintreeClient braintreeClient) {
+        this.braintreeClient = braintreeClient;
+    }
+
+    /**
+     * Forwards the user to the Samsung Pay update page.
+     * This should be invoked when Samsung Pay returns a {@link SamsungPayError#SAMSUNG_PAY_APP_NEEDS_UPDATE}
+     * error code from {@link SamsungPayClient#isReadyToPay(SamsungPayIsReadyToPayCallback)}
+     */
+    public void goToUpdatePage() {
+        getInternalClient(new GetSamsungPayInternalClientCallback() {
+            @Override
+            public void onResult(@Nullable SamsungPayInternalClient internalClient, @Nullable Exception error) {
+                if (internalClient != null) {
+                    internalClient.goToSamsungPayUpdatePage();
+                    braintreeClient.sendAnalyticsEvent("samsung-pay.goto-update-page");
+                } else {
+                    // TODO: determine if we should notify an error here
+                }
+            }
+        });
+    }
+
+    /**
+     * Forwards the user to the Samsung Pay activate page.
+     * This should be invoked when Samsung Pay returns a {@link SamsungPayError#SAMSUNG_PAY_SETUP_NOT_COMPLETED}
+     * error code from {@link SamsungPayClient#isReadyToPay(SamsungPayIsReadyToPayCallback)}
+     */
+    public void activateSamsungPay() {
+        getInternalClient(new GetSamsungPayInternalClientCallback() {
+            @Override
+            public void onResult(@Nullable SamsungPayInternalClient internalClient, @Nullable Exception error) {
+                if (internalClient != null) {
+                    internalClient.activateSamsungPay();
+                    braintreeClient.sendAnalyticsEvent("samsung-pay.activate-samsung-pay");
+                } else {
+                    // TODO: determine if we should notify an error here
+                }
+            }
+        });
+    }
+
+    /**
+     * Call isReadyToPay before starting your Samsung Pay flow. isReadyToPay will call you back with the
+     * status of Samsung Pay. If if the status of Samsung Pay is anything but [SamsungPayStatus.SPAY_READY],
+     * the listener will be called back with a value of false. If the Samsung Pay callback returns an error,
+     * see {@link SamsungPayError} for a list of possible error codes
+     *
+     * @param callback {@link SamsungPayIsReadyToPayCallback}
+     */
+    public void isReadyToPay(final SamsungPayIsReadyToPayCallback callback) {
+        getSamsungPayStatus(new GetSamsungPayStatusCallback() {
+            @Override
+            public void onResult(@Nullable Integer status, @Nullable Exception samsungPayError) {
+                if (status != null) {
```

Comment:
I think we could explore that possibility yeah. The `SamsungPayError` type exists to bring the error codes into our domain. We could do some mapping to make sure we've covered all the bases. But mainly, wanted to have an `isReadyToPay` method similar to `GooglePay` and `VenmoPay`. Form-fitting SamsungPay to this pattern wasn't super straightforward. I'm up for taking another look.

Reply:
Actually - the above comment about why not to return the status code instead of null for error status codes makes sense, so leaving this as nullable is probably the right call.